### PR TITLE
Added get raw value and sorted field helper date issue

### PR DESCRIPTION
**Pull request by tjpalmer85**
Added ability to retrieve raw value from a field and resolved get field issue where JS and Python don't agree on dates.

### CODE REVIEW

File: pyseext/field_helper.py [hunk 1]
```diff
@@ -202,13 +204,17 @@ def get_field_value(self, form_cq: str, name: str) -> Any:
...
+            is_date (bool): Indicates whether the field is a date time or not. Defaults to False.
 
         Returns:
             Any: The value of the field, or None if not found.
         """
         script = self._GET_FIELD_VALUE_TEMPLATE.format(form_cq=form_cq, name=name)
         self.ensure_javascript_loaded()
-        return self._driver.execute_script(script)
+
+        value = self._driver.execute_script(script)
```
Comment (westy) [inline pyseext/field_helper.py:216]: No need for this local now.

Sorry, picky me!!

Comment (westy) [reply pyseext/field_helper.py:216]: Although saying that, doesn't this need to convert from the locale date string?

File: pyseext/field_helper.py [hunk 2]
```diff
@@ -202,13 +204,17 @@ def get_field_value(self, form_cq: str, name: str) -> Any:
         Args:
             form_cq (str): The component query that identifies the form panel in which to look for the field
             name (str): The name of the field
+            is_date (bool): Indicates whether the field is a date time or not. Defaults to False.
```
Comment (westy) [inline pyseext/field_helper.py]: Did you remove this parameter?
Don't you need it to convert to a Python date?

**Review by westy — CHANGES_REQUESTED** [review]

File: pyseext/field_helper.py (hunk 1 above)
Comment (tjpalmer85) [reply pyseext/field_helper.py:216]: Removed local. I don't think any conversion needs to be done, as the date is returned from JS in the correct format of dd/mm/yyyy. Which is the format the UI is displaying the data and when used against the check_field_value (which is the main method we use to verify data) it asserts correctly.

File: pyseext/field_helper.py (hunk 2 above)
Comment (tjpalmer85) [reply pyseext/field_helper.py]: Yes sorry should have removed this, will do now. As above I don't think we need to convert to date as it's already in the correct format. Although saying that I wonder if there are other date fields not in that format, what will happen then. Will investigate further

Comment (westy) [reply pyseext/field_helper.py]: I suppose my point is that the field value is a date not a string, so this method is not actually matching the type that the value is.

Now you have fixed the format you should be able to convert easily.
That said, it sounds like the raw value might be easier for you to represent in checks.

I don't feel that strongly about it, so happy with whatever you decide.

Cheers

Comment (tjpalmer85) [reply pyseext/field_helper.py]: Is what is returned from the JS side not a string as toLocaleString() which is used on the JS side if field is a date returns a string? Or have I misunderstood what I thought I was doing, which is highly likely :)

Comment (westy) [reply pyseext/field_helper.py]: Yeah, it is a string, but when calling getValue on a date field you'd expect a date object.
That's why we were looking to parse it into a python date before using datetime.utils

Comment (tjpalmer85) [reply pyseext/field_helper.py]: Sorry, penny finally dropped for my pea brain. Have updated so it converts back to datetime obj if date field. And like you previously mentioned can use raw value when asserting a field value

**Review by westy — APPROVED** [review]
Lookds good, thanks!

**Pull request merged**